compiler+runtime/src/cpp/jank/runtime/obj/persistent_string_sequence.cpp

```diff
@@ -0,0 +1,113 @@
+#include <jank/runtime/obj/persistent_string_sequence.hpp>
+
+namespace jank::runtime
+{
+  obj::persistent_string_sequence::static_object(obj::persistent_string_ptr s)
+    : str{ s }
+  {
+    assert(s->count() > 0);
+  }
+
+  obj::persistent_string_sequence::static_object(obj::persistent_string_ptr s, size_t i)
+    : str{ s }
+    , index{ i }
+  {
+    assert(s->count() > 0);
+  }
+
+  /* behavior::objectable */
+  native_bool obj::persistent_string_sequence::equal(object const &o) const
+  {
+    if(o.type != object_type::persistent_string_sequence)
+    {
+      return false;
+    }
+    auto const seq(expect_object<obj::persistent_string_sequence>(&o));
+    return std::equal(str->data.begin() + index,
+                      str->data.end(),
+                      seq->str->data.begin() + seq->index,
+                      seq->str->data.end());
+  }
+
+  void obj::persistent_string_sequence::to_string(fmt::memory_buffer &buff) const
+  {
+    return behavior::detail::to_string(
+      str->data.begin()
+        + static_cast<decltype(obj::persistent_string::data)::difference_type>(index),
+      str->data.end(),
+      "(",
+      ')',
+      buff);
+  }
+
+  native_persistent_string obj::persistent_string_sequence::to_string() const
+  {
+    fmt::memory_buffer buff;
+    behavior::detail::to_string(
+      str->data.begin()
+        + static_cast<decltype(obj::persistent_string::data)::difference_type>(index),
+      str->data.end(),
+      "(",
+      ')',
+      buff);
+    return { buff.data(), buff.size() };
+  }
+
+  native_hash obj::persistent_string_sequence::to_hash() const
+  {
+    return str->data.substr(index).to_hash();
```

Comment:
This is a good attempt. It's not how Clojure's hashes work though. Hashing isn't arbitrary, but rather very intentional, since it's used for light equality checking. Sequences are hashed based on whether or not they're ordered. Ordered sequences with the same contents have the same hash. Check it:

```clojure
user=> (hash '(\f \o \o))
-1629510009
user=> (hash "foo")
493551392
user=> (hash [\f \o \o])
-1629510009
```

jank has a bunch of hash functions in `hash.hpp` for this. We'll want to use the iteretor-based `ordered`, but that will require a `char` overload for `visit`. Check out the header to see what I mean. From there, you need to figure out how to hash chars.

```clojure
user=> (hash \a)
97
user=> (hash \A)
65
```

Looks like Clojure just uses the integer value.